### PR TITLE
Reconnect when no data is received for 5 minutes.

### PR DESCRIPTION
**Pull request by drichelson**
Reconnect when no data is received for 5 minutes.

### CODE REVIEW

File: ldclient/sse_client.py [hunk 1]
```diff
@@ -139,9 +151,6 @@ def parse(cls, raw):
 
             name = m.groupdict()['name']
             value = m.groupdict()['value']
-            if name == '':
```
Comment (drichelson) [inline ldclient/sse_client.py]: this wasn't actually working.

File: ldclient/sse_client.py [hunk 2]
```diff
@@ -46,7 +48,12 @@ def _connect(self):
 
         # Use session if set.  Otherwise fall back to requests module.
         requester = self.session or requests
-        self.resp = requester.get(self.url, stream=True, **self.requests_kwargs)
+        self.resp = requester.get(
+            self.url,
+            stream=True,
+            timeout=(self._connect_timeout, self._read_timeout),
```
Comment (drichelson) [inline ldclient/sse_client.py]: Setting the read_timeout here means that the request will throw a timeout error if no data is received for 5 minutes.. thus implementing heartbeat detection.

File: ldclient/streaming.py [hunk 3]
```diff
@@ -75,6 +84,6 @@ def process_message(store, requester, msg, ready):
             # noinspection PyShadowingNames
             version = payload['version']
             store.delete(key, version)
-        else:
-            log.warning('Unhandled event in stream processor: ' + msg.event)
+        elif msg.event != 'comment':
```
Comment (jkodumal) [inline ldclient/streaming.py]: change this back

File: ldclient/sse_client.py [hunk 4]
```diff
@@ -130,6 +137,11 @@ def parse(cls, raw):
         and return a Event object.
         """
         msg = cls()
+        if raw.startswith(':'):
```
Comment (jkodumal) [inline ldclient/sse_client.py]: Remove this-- not necessary.

**Comment by jkodumal** [comment]
👍

**Pull request merged**